src/Stratis.Bitcoin.Features.WatchOnlyWallet/WatchOnlyWalletManager.cs

```diff
@@ -155,11 +231,75 @@ public void ProcessTransaction(Transaction transaction, Block block = null)
                         {
                             existingTransaction.MerkleProof = new MerkleBlock(block, new[] { transactionHash }).PartialMerkleTree;
                         }
+
+                        // Update the lookup cache with the new transaction information.
+                        this.txLookup.Remove(existingTransaction.Id);
+                        this.txLookup.Add(existingTransaction.Id, existingTransaction);
                     }
 
                     this.SaveWatchOnlyWallet();
                 }
             }
+
+            this.Wallet.WatchedTransactions.TryGetValue(transactionHash.ToString(), out TransactionData existingWatchedTransaction);
+
+            if (existingWatchedTransaction != null)
+            {
+                // The transaction was previously stored, in an unconfirmed state.
+                // So now update the block hash and Merkle proof since it has
+                // appeared in a block.
+                existingWatchedTransaction.BlockHash = block.GetHash();
+                existingWatchedTransaction.MerkleProof = new MerkleBlock(block, new[] {transaction.GetHash()}).PartialMerkleTree;
+
+                // Update the lookup cache with the new transaction information.
+                this.txLookup.Remove(existingWatchedTransaction.Id);
+                this.txLookup.Add(existingWatchedTransaction.Id, existingWatchedTransaction);
+
+                this.SaveWatchOnlyWallet();
+            }
+        }
+
+        /// <summary>
+        /// Populate the transaction lookup dictionary with the current
+        /// contents of the watch only wallet's watched addresses and
+        /// transactions.
+        /// </summary>
+        private void LoadTransactionLookup()
+        {
+            var lookup = new Dictionary<uint256, TransactionData>();
+            var watchOnlyWallet = this.GetWatchOnlyWallet();
```

Comment:
You're inside the manager so you don't need a new watchOnlyWallet variable, you have `this.wallet`.